### PR TITLE
Feature/audit 0

### PR DESCRIPTION
**Pull request by steinadio**
Filling out docstrings for all bmorph scripts and converting them into numpy style.

### CODE REVIEW

File: bmorph/util/mizuroute_utils.py [hunk 1]
```diff
@@ -152,19 +210,27 @@ def find_max_r2(ds, curr_seg_flow):
...
+    
+    Parameters
     ----
-    ds: xr.Dataset
+    ds: xarray.Dataset
         contains the variable 'reference_flow' to compare
-        curr_seg_flow against and the coordinate 'seg'
-    curr_seg_flow: 
+        curr_seg_flow against and the coordinate 'seg'.
+    curr_seg_flow: numpy.array
```
Comment (arbennett) [inline bmorph/util/mizuroute_utils.py:224]: Can just be `np.array`

File: bmorph/util/mizuroute_utils.py [hunk 2]
```diff
@@ -152,19 +210,27 @@ def find_max_r2(ds, curr_seg_flow):
...
+    """
+    Searches through ds to find which seg has the smallest
+    Kullback-Leibler Divergence value with respect to curr_seg_flow. 
+    If no seg is found, min_kldiv = -1 and min_kldiv_ref_seg = -1.
+    https://en.wikipedia.org/wiki/Kullback%E2%80%93Leibler_divergence
+    
+    Parameters
     ----
-    ds: xr.Dataset
+    ds: xarray.Dataset
```
Comment (arbennett) [inline bmorph/util/mizuroute_utils.py:221]: Can be `xr.Dataset`

File: bmorph/util/mizuroute_utils.py [hunk 3]
```diff
@@ -287,26 +392,30 @@ def map_ref_sites(routed: xr.Dataset, gauge_reference: xr.Dataset,
...
-        are
-    ----
-    routed: xr.Dataset
-    gauge_reference: xr.Dataset
-    gauge_sites = None
-        if None, gauge_sites will be taken as all those listed in
-        gauge_reference
-    route_var = 'IRFroutedRunoff'
-        variable name of flows used for fill_method purposes within routed
+    Assigns segs within routed boolean 'is_gauge' idnetifiers and
```
Comment (arbennett) [inline bmorph/util/mizuroute_utils.py:395]: Make sure it's clear that `routed` is the input. Just enclose with ``. Also "idnetifiers" -> "identifiers"

File: bmorph/util/mizuroute_utils.py [hunk 4]
```diff
@@ -511,12 +636,22 @@ def map_headwater_sites(routed: xr.Dataset):
 def calculate_cdf_blend_factor(routed: xr.Dataset, gauge_reference: xr.Dataset,
                                gauge_sites=None, fill_method='kldiv'):
     """
-    calcultes the cumulative distirbtuion function blend factor based on distance
+    Calcultes the cumulative distirbtuion function blend factor based on distance
```
[the remark this answers is not in the document]
Comment (arbennett) [reply bmorph/util/mizuroute_utils.py:639]: Also `distirbtuion` -> `distribution`

**Review by arbennett — CHANGES_REQUESTED** [review]

File: bmorph/core/workflows.py [hunk 5]
```diff
@@ -10,7 +10,65 @@ def apply_annual_bmorph(raw_ts, train_ts, obs_ts,
...
+        Bins for the first time series.
+    ybins : int
+        Bins for the second time series.
+    rtol : float
+        The desired relatie tolerance of the result for KernelDensity.
+    atol : float
+        The desired absolute tolerance of the result for KernelDensity.
+    method : str
+        Method to use for mdcdedcdfm. Currently 'hist' using hist2D and 'kde'
+        using kde2D are the only supported methods.
```
[the remark this answers is not in the document]
Comment (steinadio) [reply bmorph/core/workflows.py:61]: Correct

File: bmorph/util/mizuroute_utils.py [hunk 6]
```diff
@@ -511,12 +636,22 @@ def map_headwater_sites(routed: xr.Dataset):
...
+        Contains flow timeseries data.
+    gauge_reference: xarray.Dataset
+        Contains reference flow timeseries data for the same watershed 
+        as the routed dataset.
+    gauge_sites: list, optional
+        If None, gauge_sites will be taken as all those listed in
+        gauge_reference.        
     fill_method: str
-        see map_ref_sites for full description of how fill_method works
+        Visit map_ref_sites for full description of how fill_method works.
```
[the remark this answers is not in the document]
Comment (steinadio) [reply bmorph/util/mizuroute_utils.py:654]: That makes sense. I think I was getting really into language analysis and was trying to use a verb besides "see" for disability considerations

**Pull request merged**